src/assets/wise5/directives/teacher-summary-display/dialog-guidance-teacher-summary-display.component.ts

```diff
@@ -0,0 +1,85 @@
+import { AnnotationService } from '../../services/annotationService';
+import { CommonModule } from '@angular/common';
+import { Component } from '@angular/core';
+import { ComponentState } from '../../../../app/domain/componentState';
+import { ConfigService } from '../../services/configService';
+import { CRaterIdea } from '../../components/common/cRater/CRaterIdea';
+import { CRaterService } from '../../services/cRaterService';
+import { MatCardModule } from '@angular/material/card';
+import { MatIconModule } from '@angular/material/icon';
+import { ProjectService } from '../../services/projectService';
+import { SummaryService } from '../../components/summary/summaryService';
+import { TeacherDataService } from '../../services/teacherDataService';
+import { TeacherSummaryDisplayComponent } from './teacher-summary-display.component';
+
+@Component({
+  imports: [CommonModule, MatCardModule, MatIconModule],
+  selector: 'dialog-guidance-teacher-summary-display',
+  templateUrl: 'dialog-guidance-teacher-summary-display.component.html'
+})
+export class DialogGuidanceTeacherSummaryDisplayComponent extends TeacherSummaryDisplayComponent {
+  protected ideaCountMap: Map<string, Set<number>>;
+  protected sortedIdeas: { id: string; count: number }[] = [];
+  protected topIdeas: { id: string; count: number }[] = [];
+  protected seeAllIdeas: boolean = false;
+
+  constructor(
+    protected annotationService: AnnotationService,
+    protected configService: ConfigService,
+    private cRaterService: CRaterService,
+    protected dataService: TeacherDataService,
+    protected projectService: ProjectService,
+    protected summaryService: SummaryService
+  ) {
+    super(annotationService, configService, dataService, projectService, summaryService);
+    this.ideaCountMap = new Map<string, Set<number>>();
+  }
+
+  ngOnInit() {
+    this.getLatestWork().subscribe((componentStates) => {
+      this.extractIdeasFromComponentStates(componentStates);
+      this.sortIdeas();
+    });
+  }
+
+  private extractIdeasFromComponentStates(componentStates: ComponentState[]): void {
```

Comment:
Rename to a shorter name?
```suggestion
  private extractIdeas(componentStates: ComponentState[]): void {
```